doc/introduction/circuits.rst

```diff
@@ -255,6 +257,41 @@ For example:
 
     result = circuit(0.543)
 
+Parameter Broadcasting in QNodes
+--------------------------------
+
+Some :class:`~.pennylane.QNode` circuits support the execution at multiple parameters
+simultaneously:
+
+>>> x = np.array([0.543, 1.234])
+>>> result = circuit(x)
+>>> result
+tensor([0.85616242, 0.33046511], requires_grad=True)
+
+This is called *parameter broadcasting* (as for, say, NumPy functions executed along an axis)
+or *parameter batching* (as in the application of a function to a *batch* of parameters in
+machine learning).
+
+In addition to a more flexible execution syntax, broadcasting can yield performance boosts
+compared to the separate execution of the QNode for each parameter setting. Whether or not this is
+the case depends on quite a few details, but in particular for (at most) moderately sized circuits
+(:math:`\lesssim 20` wires) with a moderate number of parameters (:math:`\lesssim 200`) executed
+on a classical simulator, one can expect to benefit from broadcasting.
+See the :class:`~.pennylane.QNode` documentation for usage details.
+
+Many standard quantum operators support broadcasting, see the corresponding attribute
+:obj:`~.pennylane.ops.qubit.attributes.supports_broadcasting` for a list. The
+:class:`~.pennylane.operation.Operator` documentation contains implementation details
+and a guide to make custom operators compatible with broadcasting.
+Broadcasting can be used with any device, but will usually only yield performance upgrades for
+classical simulator devices like ``"default.qubit"`` that indicate that they support it:
```

Comment:
```suggestion
Broadcasting can be used with any device, but will usually only yield performance upgrades for
devices like ``"default.qubit"`` that indicate that they support it:
```

I have a feeling that even non-simulator devices will see speedups, partly due to avoiding slow python for loops.

Reply:
Ah, yes, in the QNode documentation I mentioned this but here I do not explicitly mention it, so indeed it's good to not restrict the statement, I think :)